packages/python/javdb_platform/db_layer/operations_repo.py

```
@@ -23,8 +53,170 @@ def _normalize_inventory_entry(entry: dict) -> Tuple[str, str, str, str, int, in
+    params = []
+    for year in normalized_years:
+        where_parts.append("(FolderPath = ? OR FolderPath LIKE ?)")
+        params.extend([year, f"{year}/%"])
+
+    main_cols = (
+        "VideoCode, SensorCategory, SubtitleCategory, FolderPath, "
+        "FolderSize, FileCount, DateTimeScanned"
+    )
+    statements = [
+        (
+            "DELETE FROM RcloneInventory WHERE " + " OR ".join(where_parts),
```

Ruff (0.15.12)

[error] Possible SQL injection vector through string-based query construction

(S608)